ibm/service/db2/resource_ibm_db2_instance.go

```diff
@@ -1286,6 +1385,72 @@ func resourceIBMDb2InstanceCreate(d *schema.ResourceData, meta interface{}) erro
 		}
 	}
 
+	if allowlistConfigRaw, ok := d.GetOk("allowlist_config"); ok {
+		list := allowlistConfigRaw.([]interface{})
+		if len(list) == 0 {
+			fmt.Println("No allowlist config provided, skipping.")
+		} else {
+			ipList := make([]db2saasv1.IpAddress, 0)
+
+			for _, item := range list {
+				entry := item.(map[string]interface{})
+
+				if ipAddrsRaw, ok := entry["ip_addresses"]; ok {
+					ipAddrs := ipAddrsRaw.([]interface{})
+					for _, ipItem := range ipAddrs {
+						ipEntry := ipItem.(map[string]interface{})
+						var address, description string
+
+						if rawAddress, ok := ipEntry["address"]; ok && rawAddress != nil {
+							str, ok := rawAddress.(string)
+							if !ok {
```

Comment:
These block are explictly n't required since the schema we defined for arguments handles

Reply:
@hkantare I'm keeping the ip check, as without that provisioning goes through, but user won't be able to know what went wrong and due to retries it is increasing time for provisining.
Until and unless user enables debug log, he/she won't be able to know what went wrong.

But with IP check, it is early exit.